File: python/src/Structural/Bridge/Soda.py

from SodaImpSingleton import SodaImpSingleton

class Soda:

	def __init__(self) :
		raise Exception("Can't instantiate abstract class, Soda")

	def setSodaImp(self) : 
		self.sodaImp = SodaImpSingleton._sodaimp

	def getSodaImp(self) :
		return self.sodaImp

	def pourSoda(self) : 
		raise Exception("Unimplemented method Soda.pourSoda")

File: ruby/src/Structural/Bridge/Soda.rb

require "Structural/Bridge/SodaImpSingleton"

class Soda

	def initialize()
		raise Exception("Can't instantiate abstract class, Soda")
	end

	def setSodaImp()
 		@sodaImp = SodaImpSingleton.instance.getTheSodaImp
	end

	def getSodaImp()
 		return @sodaImp
	end

	def pourSoda()
 		raise Exception("Unimplemented method Soda.pourSoda")
	end

end

